### PR TITLE
Split state into distinct fields

### PR DESCRIPTION
**Pull request by Salazander**
## Description
Instead of one convoluted string to describe the state, use 2 enums and 1 string, each with a very specific meaning

### CODE REVIEW

**Review by MagdaPaj — APPROVED** [review]
Looks very good!

**Review by syedhassaanahmed — APPROVED** [review]
🚀

File: src/modules/packagemanagerconfiguration/README.md [hunk 1]
```diff
@@ -34,31 +34,28 @@ The state of the sources  is reported by listing all files from `/etc/apt/source
...
+The `ExecutionState`, `ExecutionSubState` and `ExecutionSubStateDetails` properties reflect the current state and can have the following values:
+
+| ExecutionState | ExecutionSubState             | ExecutionSubStateDetails | Meaning                                                                                      |
+| -------------- |------------------------------ | ------------------------ | -------------------------------------------------------------------------------------------- |
+| 0 (Unknown)    | 0 (None)                      | empty                    | No desired properties are known to the module. This is the initial default state.            |
+| 1 (Running)    | 1 (DeserializingJsonPayload)  | empty                    | Deserializing PackageManagerConfiguration JSON object                                        |
+| 1 (Running)    | 2 (DeserializingDesiredState) | empty                    | Deserializing DesiredState JSON object                                                       |
+| 1 (Running)    | 3 (DeserializingSources)      | {source_name}            | Deserializing Sources JSON object                                                            |
```
Comment (MagdaPaj) [inline src/modules/packagemanagerconfiguration/README.md:44]: Nit: maybe deleting the curly brackets from `ExecutionSubStateDetails? before we had them in the executionState string, now not anymore.

**Review by Ioana37 — APPROVED** [review]

File: src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfiguration.cpp [hunk 2]
```diff
@@ -169,7 +170,8 @@ int PackageManagerConfigurationBase::Set(const char* componentName, const char*
 
                         if (status == MMI_OK)
                         {
-                            status = PackageManagerConfigurationBase::ExecuteUpdates(desiredState.packages);
+                            bool updatePackageLists = desiredState.sources.empty();
```
Comment (MagdaPaj) [inline src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfiguration.cpp:173]: nit: could we use this variable also in line 166?

File: src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfiguration.cpp [hunk 3]
```diff
@@ -382,17 +384,21 @@ int PackageManagerConfigurationBase::ExecuteUpdate(const std::string &value)
...
 
-int PackageManagerConfigurationBase::ExecuteUpdates(const std::vector<std::string> packages)
+int PackageManagerConfigurationBase::ExecuteUpdates(const std::vector<std::string> packages, bool updatePackageLists)
 {
     int status = MMI_OK;
-    m_executionState.SetExecutionState(StateComponent::Running, SubStateComponent::UpdatingPackagesLists);
 
-    status = RunCommand(g_commandAptUpdate.c_str(), true, nullptr, 0);
-    if (status != MMI_OK)
+    if(updatePackageLists)
```
Comment (MagdaPaj) [inline src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfiguration.cpp]: nit: missing space

**Review by MagdaPaj — APPROVED** [review]

**Pull request merged**